http/middleware.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/** Options for {@linkcode MiddlewareHandler}. */
+export interface MiddlewareOptions<T = undefined> {
+  /** Calls the next middleware handler in the middleware chain. */
+  next: () => Response | Promise<Response>;
+  /** State that gets passed down through each middleware handler. */
+  state: T;
+}
+
+/**
+ * Middleware handler which extends {@linkcode Deno.ServeHandlerInfo}. Used
+ * in {@linkcode composeMiddleware}.
+ */
+export type MiddlewareHandler<T = undefined> = (
+  request: Request,
+  info: Deno.ServeHandlerInfo,
```

Comment:
I know this is from the Deno.serve API, but might it be cleaner to move it into either the options object or the options state object?
1. It follows the deno_std guidelines "max 2 args, put the rest into an options object"
2. Deno.ServeHandlerInfo isn't a very interesting object, no offence 😅, but it adds noise to the API when I can imagine most people will never use it.

Reply:
Yeah, I'm thinking that too.